x/deployment/query/types.go

```diff
@@ -18,9 +18,11 @@ type DeploymentFilters struct {
 	State types.DeploymentState
 }
 
+// Accept returns whether deployment filters valid or not
 func (filters DeploymentFilters) Accept(obj types.Deployment, isValidState bool) bool {
-	if (filters.Owner.Empty() && !isValidState) ||
-		(filters.Owner.Empty() && (obj.State == filters.State)) ||
+	err := sdk.VerifyAddressFormat(filters.Owner)
```

Comment:
we need to test all of these query filters thoroughly; maybe leave it out of this PR.

I think that this isn't the semantics we want - if it's an empty address then don't filter on the address, if it's a _bad_ address then return error.